src/tools/wasm-emscripten-finalize.cpp

```diff
@@ -175,10 +178,9 @@ int main(int argc, const char* argv[]) {
 
   options.applyFeatures(wasm);
 
-  if (options.debug) {
-    std::cerr << "Module before:\n";
-    WasmPrinter::printModule(&wasm, std::cerr);
-  }
+  // Useful for debugging
```

Comment:
It feels weird to go from a runtime check to a compile-time thing here. Maybe add a new runtime option here? `--dump-module`?
I do agree that I usually don't want a pre-module dump, so I do support decoupling debug from dump here.

Also we should update the "Module after" dump to BYN_TRACE + BYN_DEBUG too? (Line293, can't comment it directly)

Reply:
Done